packages/jazz-tools/src/tests/schema.test.ts

```diff
@@ -163,6 +163,35 @@ describe("co.json TypeScript validation", () => {
       }>();
   });
 
+  /* Special case from reported issue:
+   ** See: https://github.com/garden-co/jazz/issues/1496
+   */
+  it("should apply the same validation to optional json [JAZZ-1496]", async () => {
+    interface ValidInterface0 {
+      value: string;
+    }
+    interface ValidInterface1 {
+      value: string | undefined;
+    }
+    interface InterfaceWithOptionalTypes {
+      value?: string;
+    }
+
+    class MapWithOptionalJSON extends CoMap {
+      data1 = co.optional.json<ValidInterface0>();
+      data2 = co.optional.json<ValidInterface1>();
+      data3 = co.optional.json<InterfaceWithOptionalTypes>();
```

Comment:
Noticed that the CI wasn't typechecking the tests on jazz-tools.

Updated the tsconfig and now it reports that there is an indeed an issue on this type:
```
ype 'InterfaceWithOptionalTypes' does not satisfy the constraint 'CoJsonValue<InterfaceWithOptionalTypes>'.
  Type 'InterfaceWithOptionalTypes' is not assignable to type '{ value: CoJsonValue1L<string | undefined> | undefined; }'.
    Property 'value' is optional in type 'InterfaceWithOptionalTypes' but required in type '{ value: CoJsonValue1L<string | undefined> | undefined; }'
```